File: 338-counting-bits/338-counting-bits.cpp
class Solution {
public:
    int getBit(int n){
        if(n==0){
            return 0;
        }
        return (n&1)+getBit(n>>1);
    }
    vector<int> countBits(int n) {
        vector<int> ans;
        for(int i=0;i<=n;i++){
            ans.push_back(getBit(i));
        }
        return ans;
    }
};

File: 338-counting-bits/338-counting-bits.js
function getBit(n){
    let count = 0;
    while(n){
        count+=(n&1);
        n = n>>1;
    }
    return count;
}
/**
 * @param {number} n
 * @return {number[]}
 */
var countBits = function(n) {
    let ans = [];
    for(let i=0;i<=n;i++){
        ans.push(getBit(i));
    }
    return ans;
};